next-app/src/lib/fetch/server.js

```diff
@@ -0,0 +1,63 @@
+import { throwError } from '$lib/error'
+
+/**
+ *
+ * @typedef RPC
+ * @type {object}
+ * @property {string} name Name of the remote procedure to call
+ * @property {string[]} [args] Arguments to pass to the remote procedure
+ * @property {string} [cookie] https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie
+ *
+ * @param { RPC } rpc
+ */
+export async function sf(rpc) {
+	const { name, args = [], cookie } = rpc
+
+	const body = {
+		id: Date.now(),
+		method: name,
+		params: {
+			orderedParams: args,
+		},
+	}
+
+	const results = await customFetch(`${process.env.API_HOST}/api/sf`, 'post', body, cookie)
+
+	if (results.error) {
+		throwError(results.error.message, 500)
+	}
+
+	if (! results.result) {
+		throwError('Badly formed response, missing result', 500)
+	}
+
+	return results.result
+}
+
+async function customFetch(url, method, body, cookie) {
+	let response = {}
+
+	try {
+		// https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Supplying_request_options
+		response = await fetch(url, {
+			method,
+			// credentials: 'include', // ensures the response back from the api will be allowed to "set-cookie"
+			headers: {
+				'content-type': 'application/json',
+				cookie,
+			},
+			body: JSON.stringify(body),
+		})
+	} catch {
+		// these only occur for network errors, like these:
+		//	request made with a bad host, e.g., //httpbin
+		//	the host is refusing connections
+		throwError('NETWORK ERROR', 500)
```

Comment:
At least for the deployment to QA, I think I'd like this to be a slightly more informative error message, i.e. `catch (e)` and then have `throwError` take an optional third parameter that would be the original exception. We might have code in `throwError` that only passes on that exception if the environment is "dev" or "qa", since we don't want to expose possibly-sensitive exception info on prod. But since I expect that the precise exception thrown would be useful in diagnosing any network errors, I'd like to keep it around in some form unless we're in prod.

Reply:
we may need to discuss this more... the exception block you noted here wouldn't get tripped for a `500` from the api call... we wouldn't know about it until we inspected the response.  If it were a 500 we'd need to take a closer look at the shape of the data in that response.  As far as I could tell the only valuable data from `sf` failures was in the fields I pass along...

I did not shut the db down to see what the error response looks like in that case, maybe it's diff.  If so, we'll need to potentially look it various places for the error info.

Reply:
fwiw, shutting the db down locally and trying a change password results in the following:
![image](https://user-images.githubusercontent.com/4412848/161832785-7ad8169f-adba-49e4-b410-f447f75a0ce0.png)